### PR TITLE
Functional testsuite restructure

### PR DESCRIPTION
**Pull request by diego-G**
Set the ground for functional test

Subtask of #678

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "test": "./node_modules/.bin/grunt test --verbose",
     "test-unit": "./node_modules/.bin/grunt test-unit --verbose",
     "test-integration": "grunt test-integration",
-    "test-functional": "grunt test-functional",
+		"test-functional-ws": "grunt test-functional-ws",
```
Comment (Isabello) [inline package.json]: fix me

File: test/common/complexTransactions.js [hunk 2]
```diff
@@ -1,32 +1,67 @@
...
 }
 
 function sendSignature (signature, transaction, cb) {
-	http.post('/api/signatures', { signature: signature, transaction: transaction.id }, function (err, res) {
-		if (res.body.success) {
-			return cb(null, res.body);
-		}
-		return cb(res.body);
-	});
+	http.post('/api/signatures', {signature: {signature: signature, transaction: transaction.id}}, httpCallbackHelper.bind(null, cb));
```
Comment (Isabello) [inline test/common/complexTransactions.js]:
```
{ 
   signature: {
     signature: signature,
     transaction: transactionId
   }
}
```
Is this correct?

Comment (diego-G) [reply test/common/complexTransactions.js]: It is.

Comment (Isabello) [reply test/common/complexTransactions.js]: ok! looks weird to me :) the new api has it like this:

```
{
  "signatures": [
    {
      "transactionId": "222675625422353767",
      "signature": "2821d93a742c4edf5fd960efad41a4def7bf0fd0f7c09869aed524f6f52bf9c97a617095e2c712bd28b4279078a29509b339ac55187854006591aa759784c205"
    }
  ]
}
```

Comment (diego-G) [reply test/common/complexTransactions.js]: Your guess is right. It is weird. But for now it works like that. We will adjust the code on the API refactoring.

File: test/node.js [hunk 3]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
 	return parseInt(node.fees.transactionFee) + parseInt(node.fees.dataFee);
 };
 
-// Returns a random username
+// Returns a random username of 16 characters
 node.randomUsername = function () {
```
Comment (MaciejBaj) [inline test/node.js]: True random username would be dependent also from different length. Can you do sth like merge between letter part - `randomString.generate({length: node.randomNumber(1, 8), charset: 'alphabetic')`
and nonletter - `randomString.generate({length: node.randomNumber(1, 8), charset: '0123456789!@$&_.')`

File: test/node.js [hunk 4]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
+		charset: custom
+	});
 
-	return username;
+	return randomLetter.concat(username);
 };
 
-// Returns a random capitialized username
+// Returns a random capitialized username of 16 characters
 node.randomCapitalUsername = function () {
```
Comment (MaciejBaj) [inline test/node.js]: same for capital one

File: test/node.js (hunk 3 above)
Comment (diego-G) [reply test/node.js]: Done! Thanks @MaciejBaj

File: test/node.js [hunk 5]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
-// Returns a random username
+// Returns a random username of 16 characters
 node.randomUsername = function () {
-	var size = node.randomNumber(1, 16); // Min. username size is 1, Max. username size is 16
-	var username = '';
-	var possible = 'abcdefghijklmnopqrstuvwxyz0123456789!@$&_.';
+	var randomLetter = randomString.generate({
+	  length: 1,
+	  charset: 'alphabetic',
+		capitalization: 'lowercase'
```
Comment (MaciejBaj) [inline test/node.js]: sth with indent here

File: test/node.js [hunk 6]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
-		username += possible.charAt(Math.floor(Math.random() * possible.length));
-	}
+	return randomLetter.concat(username);
+};
+
+// Returns a random delegate name of 20 characters
+node.randomDelegateName = function () {
+	var randomLetter = randomString.generate({
+	  length: 1,
+	  charset: 'alphabetic',
```
Comment (MaciejBaj) [inline test/node.js]: sth with indent here

File: test/node.js [hunk 7]
```diff
@@ -341,6 +363,13 @@ node.randomTxAccount = function () {
 	});
 };
 
+// Returns an random basic transaction to send 1 LSK from genesis account to a random account
+node.randomTx = function (offset) {
+	var randomAccount = node.randomAccount();
+
```
Comment (MaciejBaj) [inline test/node.js]: Remove empty line

Comment (diego-G) [reply test/node.js]: We always leave one empty line between the definition of variables and execution/return.

File: test/node.js (hunk 5 above)
Comment (diego-G) [reply test/node.js]: Done.

File: test/common/complexTransactions.js [hunk 8]
```diff
@@ -1,32 +1,67 @@
...
 }
 
 function sendLISK (params, cb) {
-	sendTransaction(lisk.transaction.createTransaction(params.address, params.amount, params.secret, params.secondSecret), cb);
+	console.log(params);
```
Comment (4miners) [inline test/common/complexTransactions.js]: Why `console.log`? There is proper function already.

File: test/functional/http/get/multisignatures.js [hunk 9]
```diff
@@ -118,7 +119,8 @@ describe('POST signatures/sign (regular account)', function () {
 				node.expect(res.body.transaction).to.have.property('id').to.equal(transaction.id);
 				var signature = node.lisk.multisignature.signTransaction(transaction, multisigAccount.password);
 				sendSignature(signature, transaction, function (err, res) {
-					node.expect(err).not.to.be.empty;
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Error processing signature: Transaction not found');
```
Comment (4miners) [inline test/functional/http/get/multisignatures.js]: Description, why `should be impossible to sign the transaction`?

File: test/functional/http/get/peers.js [hunk 10]
```diff
@@ -3,12 +3,12 @@
...
-var Peer = require('../../logic/peer.js');
-var peersSortFields = require('../../sql/peers').sortFields;
-var wsServer = require('../common/wsServer');
-var testConfig = require('../config.json');
+var node = require('../../../node.js');
+var http = require('../../../common/httpCommunication.js');
+var Peer = require('../../../../logic/peer.js');
+var peersSortFields = require('../../../../sql/peers').sortFields;
+var wsServer = require('../../../common/wsServer');
+var testConfig = require('../../../config.json');
```
Comment (4miners) [inline test/functional/http/get/peers.js]: Shouldn't we use config from `node.`?

File: test/node.js [hunk 11]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
+// Returns a random delegate name of 20 characters
+node.randomDelegateName = function () {
+	var randomLetter = randomString.generate({
+		length: 1,
+		charset: 'alphabetic',
+		capitalization: 'lowercase'
+	});
+	var custom = 'abcdefghijklmnopqrstuvwxyz0123456789!@$&_.';
+	var username = randomString.generate({
+		length: node.randomNumber(1,19),
```
Comment (4miners) [inline test/node.js]: space (`1, 19`)

File: test/node.js [hunk 12]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
-	var username = '';
-	var possible = 'abcdefghijklmnopqrstuvwxyz0123456789!@$&_.';
+	var randomLetter = randomString.generate({
+		length: 1,
+		charset: 'alphabetic',
+		capitalization: 'lowercase'
+	});
+	var custom = 'abcdefghijklmnopqrstuvwxyz0123456789!@$&_.';
+	var username = randomString.generate({
+		length: node.randomNumber(1,15),
```
Comment (4miners) [inline test/node.js]: same

File: test/node.js [hunk 13]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
-		username += possible.charAt(Math.floor(Math.random() * possible.length));
-	}
+	var randomLetter = randomString.generate({
+		length: 1,
+		charset: 'alphabetic',
+		capitalization: 'uppercase'
+	});
+	var custom = 'abcdefghijklmnopqrstuvwxyz0123456789!@$&_.';
+	var username = randomString.generate({
+		length: node.randomNumber(1,15),
```
Comment (4miners) [inline test/node.js]: same

File: test/node.js [hunk 14]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
-	var size = node.randomNumber(1, 32); // Min. username size is 1, Max. username size is 32
-	var name = 'A';
-	var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+	var custom = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
 
-	for (var i = 0; i < size - 1; i++) {
-		name += possible.charAt(Math.floor(Math.random() * possible.length));
-	}
+	return randomString.generate({
+		length: node.randomNumber(1,32),
```
Comment (4miners) [inline test/node.js]: same

File: test/node.js [hunk 15]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
+		charset: custom
+	});
+};
+
+// Test random application
+node.randomApplication = function () {
+	var application = {
+		category: node.randomNumber(0,9),
```
Comment (4miners) [inline test/node.js]: space

File: test/node.js [hunk 16]
```diff
@@ -276,43 +264,77 @@ node.expectedFeeForTrsWithData = function (amount) {
...
+		name: node.randomApplicationName(),
+		description: 'Blockchain based home monitoring tool',
+		tags: 'monitoring temperature power sidechain',
+		type: node.randomNumber(0,2),
```
Comment (4miners) [inline test/node.js]: space

File: test/functional/http/get/multisignatures.js (hunk 9 above)
Comment (diego-G) [reply test/functional/http/get/multisignatures.js]: We need to rewrite these tests. They are not enough. The purpose of this PR is not that.

File: test/functional/http/get/peers.js (hunk 10 above)
Comment (diego-G) [reply test/functional/http/get/peers.js]: @4miners in order to do that, we need to wait for #717

File: test/node.js (hunk 11 above)
Comment (diego-G) [reply test/node.js]: Please @4miners , next time add a commit yourself addressing this purposes. We don't have a proper Eslint configuration which defines a style.

**Pull request merged**